pages/rops/procedures/list/views/index.jsx

```diff
@@ -0,0 +1,153 @@
+};
+
+const formatters = {
+  species: {
+    format: s => {
+      const species = allSpecies.find(as => as.value === s);
+      return species ? species.label : s;
+    }
+  },
+  reuse: {
+    format: yn
+  },
+  placesOfBirth: {
+    format: getRadioOption('placesOfBirth')
+  },
+  nhpsOrigin: {
+    format: getRadioOption('nhpsOrigin')
+  },
+  nhpsColonyStatus: {
+    format: getRadioOption('nhpsColonyStatus')
+  },
+  nhpsGeneration: {
+    format: getRadioOption('nhpsGeneration')
+  },
+  ga: {
+    format: getRadioOption('ga')
+  },
+  newGeneticLine: {
+    format: yn
+  },
+  purposes: {
+    format: getRadioOption('purposes')
+  },
+  subpurpose: {
+    format: (v, model) => {
+      switch (model.purposes) {
+        case 'basic':
+          return getRadioOption('basicSubpurposes')(model.basicSubpurposes);
+        case 'regulatory':
+          return getRadioOption('regulatorySubpurposes')(model.regulatorySubpurposes);
+        case 'translational':
+          return getRadioOption('translationalSubpurposes')(model.translationalSubpurposes);
+        default:
+          return '-';
+      }
+    }
+  },
+  regulatoryLegislation: {
+    format: getRadioOption('regulatoryLegislation')
+  },
+  severity: {
+    format: getRadioOption('severity')
+  }
+};
+
+function Actions({ model }) {
+  const url = getUrl({ page: 'rops.procedures.update', procedureId: model.id });
+
+  function onClick(e) {
+    if (window.confirm('Are you sure you want to delete this procedure?')) {
+      return true;
+    }
+    e.preventDefault();
+  }
+
+  return (
+    <Fragment>
+      <Link page="rops.procedures.update" procedureId={model.id} label="Edit" />
+      <form method="POST" action={`${url}/delete`}>
+        <button className="link" onClick={onClick}>Delete</button>
+      </form>
+    </Fragment>
+  );
+}
+
+export default function Procedures() {
+  const { project, rop } = useSelector(state => state.static);
+  const editable = rop.status === 'draft';
+  console.log({ editable });
+  return (
+    <Fragment>
+      <DocumentHeader
+        title={<Snippet>title</Snippet>}
+        subtitle={project.title}
+      />
+
+      <h2><Snippet canEdit={editable}>change.title</Snippet></h2>
+      <p><Snippet canEdit={editable}>change.content</Snippet></p>
+      <Details summary={<Snippet>change.summary</Snippet>}>
+        <WidthContainer>
+          <Confirm />
+        </WidthContainer>
+      </Details>
+      <br />
+
+      <h2><Snippet>procedures.title</Snippet></h2>
+      {
+        editable && (
+          <Fragment>
+            <p><Snippet>procedures.content</Snippet></p>
+            <Link
+              className="govuk-button"
+              page="rops.procedures.create"
+              label={<Snippet>procedures.add</Snippet>}
+            />
+          </Fragment>
+        )
+      }
+      {
+        rop.procedures.length
+          ? (
+            <div style={{ overflowX: 'scroll', maxWidth: '2000px' }}>
```

Comment:
What's the source of this magic number?

Reply:
Am I right in thinking that this caps the total width of the table at 2000px even if the user is viewing on a super-wide monitor?

Are we sure that's a good thing? Letting it go to viewport width however large that might be seems potentially better

Reply:
not actually sure, let me try without